components/private_channel/browser/private_channel.cc

```diff
@@ -0,0 +1,298 @@
+  resource_request->url = GURL(request_utils::BuildUrl(
+      request_utils::EndpointType::FIRST_ROUND, PRIVATE_CHANNEL_API_VERSION));
+  std::string content_type = "application/x-www-form-urlencoded";
+  resource_request->headers.SetHeader("Content-Type", content_type);
+  resource_request->load_flags =
+      net::LOAD_DO_NOT_SEND_COOKIES | net::LOAD_DO_NOT_SAVE_COOKIES |
+      net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE |
+      net::LOAD_DO_NOT_SEND_AUTH_DATA;
+
+  network::mojom::URLLoaderFactory* loader_factory =
+      g_browser_process->system_network_context_manager()
+          ->GetURLLoaderFactory();
+
+  net::NetworkTrafficAnnotationTag traffic_annotation =
+      net::DefineNetworkTrafficAnnotation("brave_private_channel_first_round",
+                                          R"(
+        semantics {
+          sender:
+            "Brave Private Channel Service"
+          description:
+            "Runs first round of Private Channel protocol"
+          trigger:
+            "When running attestation based on Private Channels"
+          destination: WEBSITE
+        }
+        policy {
+          cookies_allowed: NO
+          setting:
+            "This feature cannot be disabled by settings."
+          policy_exception_justification:
+            "Not implemented."
+        })");
+
+  http_loader_ = network::SimpleURLLoader::Create(std::move(resource_request),
+                                                  traffic_annotation);
+  http_loader_->SetAllowHttpErrorResults(true);
+  http_loader_->AttachStringForUpload(payload, content_type);
+  http_loader_->DownloadToString(
+      loader_factory,
+      base::BindOnce(&PrivateChannel::OnPrivateChannelFirstRoundLoadComplete,
+                     base::Unretained(this), request_artefacts.client_sks,
+                     referral_code_, request_artefacts.encrypted_hashes_size),
+      kMaxPrivateChannelServerResponseSizeBytes);
+}
+
+void PrivateChannel::OnPrivateChannelFirstRoundLoadComplete(
+    std::string client_sks,
+    std::string id,
+    int encrypted_hashes_size,
+    std::unique_ptr<std::string> response_body) {
+  int response_code = -1;
+  if (http_loader_->ResponseInfo() && http_loader_->ResponseInfo()->headers)
+    response_code = http_loader_->ResponseInfo()->headers->response_code();
+
+  const std::string safe_response_body =
+      response_body ? *response_body : std::string();
+
+  if (http_loader_->NetError() != net::OK || response_code < 200 ||
+      response_code > 299) {
+    LOG(ERROR)
+        << "Failed to run the first round of the private channels protocol"
+        << ", error: " << http_loader_->NetError()
+        << ", response code: " << response_code
+        << ", payload: " << safe_response_body
+        << ", url: " << http_loader_->GetFinalURL().spec();
+    return;
+  }
+
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
+      FROM_HERE,
+      base::Bind(&PrivateChannel::SecondRoundProtocol, base::Unretained(this),
```

Comment:
passing `base::Unretained` here is not safe (after the memory leak creating PrivateChannel is fixed) because the task runner can outlive PrivateChannel

Reply:
PrivateChannel should own the task runner